src/components/Pages/Form/Form.tsx

```diff
@@ -127,22 +155,27 @@ const HabitForm = () => {
             }}
           >
             <Typography variant="body1">Rewards:</Typography>
-            {rewards.map((reward) => (
-              <Chip
-                label={reward.label}
-                key={reward.id}
-                onDelete={() =>
-                  setRewards(
-                    rewards.filter(
-                      (rewardToDelete) => rewardToDelete.id !== reward.id
-                    )
-                  )
-                }
-              />
-            ))}
+            {formik.values.rewards.length > 0 &&
+              formik.values.rewards.map(
+                (reward: { label: string; id: number }) => (
+                  <Chip
+                    label={reward.label}
+                    key={reward.id}
+                    onDelete={() =>
+                      formik.setFieldValue(
+                        "rewards",
+                        formik.values.rewards.filter(
+                          (rewardToDelete: { id: number; label: string }) =>
```

Comment:
The typing should not be needed here. I hope that `formik.values.rewards` is already typed. Or isn't it?